lib/spack/spack/test/packaging.py

```diff
@@ -219,8 +233,8 @@ def test_packaging(mock_archive, tmpdir):
 
 def test_relocate():
     assert (needs_binary_relocation('relocatable') is False)
```

Comment:
Could you move these into a separate unit test and add examples for elf/mach-o files? I'm thinking something like the following:

```
def test_needs_relocation():
    binary_type = (
        'ELF 64-bit LSB executable, x86-64, version 1 (SYSV),'
        ' dynamically linked (uses shared libs),'
        ' for GNU/Linux x.y.z, stripped')

    assert needs_binary_relocation(binary_type, os_id='Linux')
    assert not needs_binary_relocation('relocatable')
    assert not needs_binary_relocation('symbolic link to `foo\'')

    assert needs_text_relocation('ASCII text')
    assert not needs_text_relocation('symbolic link to `foo.text\'')
```

I think this would work better if you modified the `needs_binary_relocation` method to take an optional sys_name parameter (so that this test wouldn't fail on Mac OS):

`needs_binary_relocation(filetype, os_id=None):`

(you could set os_id yourself if it is not set, the only caller that would set that parameter would be the unit test).

Reply:
Sure I will get to this tomorrow.